### PR TITLE
[AAE-19610] Add specific date range settings to search filter

### PR DESCRIPTION
**Pull request by DiogoABastos**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
Search filter was missing important settings for `date-range` type

**What is the new behaviour?**
Important settings for `date-range` type were introduced to fix this bug:
https://alfresco.atlassian.net/browse/AAE-19610

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Review by eromano — APPROVED** [review]

File: lib/content-services/src/lib/search/components/search-date-range-tabbed/search-date-range-tabbed.component.ts [hunk 1]
```diff
@@ -163,9 +172,13 @@ export class SearchDateRangeTabbedComponent implements SearchWidget, OnInit {
...
             if (displayValue) {
-                const displayLabelForField = `${this.translateService.instant(this.settings.displayedLabelsByField[fieldForDisplayLabel]).toUpperCase()}: ${displayValue}`;
+                const displayLabelForField = `${this.translateService.instant(this.getDisplayLabelForField(fieldForDisplayLabel)).toUpperCase()}: ${displayValue}`;
                 this.combinedDisplayValue = this.combinedDisplayValue ? `${this.combinedDisplayValue} ${displayLabelForField}` : `${displayLabelForField}`;
             }
         });
     }
+
+    private getDisplayLabelForField(fieldForDisplayLabel: string): string {
+        return this.settings && this.settings.displayedLabelsByField && this.settings.displayedLabelsByField[fieldForDisplayLabel] ? this.settings.displayedLabelsByField[fieldForDisplayLabel] : fieldForDisplayLabel;
```
Comment (MichalKinas) [inline lib/content-services/src/lib/search/components/search-date-range-tabbed/search-date-range-tabbed.component.ts:182]: Can be way shorter like: `return this.settings?.displayedLabelsByField?.[fieldForDisplayLabel] ?? fieldForDisplayLabel;`

File: lib/content-services/src/lib/search/components/search-date-range-tabbed/search-date-range-tabbed.component.ts [hunk 2]
```diff
@@ -67,6 +69,13 @@ export class SearchDateRangeTabbedComponent implements SearchWidget, OnInit {
 
     ngOnInit(): void {
         this.fields = this.settings?.field.split(',').map(field => field.trim());
+        this.setDefaultDateFormatSettings();
+    }
+
+    private setDefaultDateFormatSettings() {
+        if (this.settings && !this.settings.dateFormat) {
```
Comment (MichalKinas) [inline lib/content-services/src/lib/search/components/search-date-range-tabbed/search-date-range-tabbed.component.ts:76]:
```suggestion
        if (!this.settings?.dateFormat) {
```

File: lib/content-services/src/lib/pipes/tab-labels.pipe.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import { Pipe, PipeTransform } from '@angular/core';
+import { SearchWidgetSettings } from '../search/models/search-widget-settings.interface';
+
+@Pipe({
+  name: 'tabLabels'
+})
+export class TabLabelsPipe implements PipeTransform {
+
+  transform(field: string, settings?: SearchWidgetSettings): string {
+    return settings && settings.displayedLabelsByField && settings.displayedLabelsByField[field] ? settings.displayedLabelsByField[field] : field;
```
Comment (MichalKinas) [inline lib/content-services/src/lib/pipes/tab-labels.pipe.ts:27]: Can be way shorter like: return this.settings?.displayedLabelsByField?.[field] ?? field;

**Review by MichalKinas — COMMENTED** [review]
Sorry that I reviewed it a bit late, I couldn't do it earlier, overall PR looks okay I only pointed out some small improvements that can be added in other PR

**Pull request merged**